### PR TITLE
Downgrade the AWS SDK to v2

### PR DESCRIPTION
**Pull request by lowjoel**
Closes https://github.com/backstage/backstage/issues/4415

This allows interoperability with AWS EKS IAM Role-linked Service
Accounts.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: packages/techdocs-common/src/stages/publish/awsS3.ts [hunk 1]
```diff
@@ -142,12 +145,12 @@ export class AwsS3Publish implements PublisherBase {
         const entityRootDir = `${entity.metadata.namespace}/${entity.kind}/${entity.metadata.name}`;
         const destination = `${entityRootDir}/${relativeFilePath}`; // S3 Bucket file relative path
 
-        const fileContent = await fs.readFile(filePath, 'utf8');
+        const fileStream = await fs.createReadStream(filePath);
```
Comment (lowjoel) [inline packages/techdocs-common/src/stages/publish/awsS3.ts:148]: This also prevents buffering the entire file contents in memory.

Comment (OrkoHunter) [reply packages/techdocs-common/src/stages/publish/awsS3.ts]: Nice!

Comment (lowjoel) [reply packages/techdocs-common/src/stages/publish/awsS3.ts]: except that it seems to break the tests 😢

Comment (OrkoHunter) [reply packages/techdocs-common/src/stages/publish/awsS3.ts]: The AWS library is mocked inside `techdocs-common/__mocks__`. I think we need to update that as well for the tests. 😄 

https://github.com/backstage/backstage/pull/3953/files

File: packages/techdocs-common/src/stages/publish/awsS3.test.ts [hunk 2]
```diff
@@ -102,6 +102,12 @@ describe('AwsS3Publish', () => {
     });
```
Comment (OrkoHunter) [inline packages/techdocs-common/src/stages/publish/awsS3.test.ts:102]: Is the file permissions mode change needed? `100755 → 100644`

File: plugins/catalog-backend/package.json [hunk 3]
```diff
@@ -38,6 +38,7 @@
     "@octokit/graphql": "^4.5.8",
     "@types/express": "^4.17.6",
     "@types/ldapjs": "^1.0.9",
+    "aws-sdk": "^2.840.0",
```
Comment (OrkoHunter) [inline plugins/catalog-backend/package.json:41]: We should remove the V3 dependency here.

`"@aws-sdk/client-organizations": "^3.2.0"`

**Review by OrkoHunter — APPROVED** [review]
Thanks for doing this! Verified and works with TechDocs. 🎉 

Left couple of comments.

File: packages/techdocs-common/src/stages/publish/awsS3.test.ts (hunk 2 above)
Comment (Rugvip) [reply packages/techdocs-common/src/stages/publish/awsS3.test.ts:102]: Yep, we want 644 tho, so 👍

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 

Just the `@aws-sdk/client-organizations` dep that we want to get rid of right? Assuming that was added when migrating to v3

File: plugins/catalog-backend/package.json (hunk 3 above)
Comment (lowjoel) [reply plugins/catalog-backend/package.json:41]: I realise there's still the Credential Provider Node used in the Kubernetes plugin

**Comment by lowjoel** [comment]
@OrkoHunter @Rugvip Fixed the Organisations package.json. The Kubernetes plugin backend started straight out with AWS SDK v3, not sure what we should do there.

**Comment by OrkoHunter** [comment]
Let's get this in and raise an issue for the Kubernetes plugin? cc @mclarke47

**Review by Rugvip — APPROVED** [review]
Yep let's treat it separately, :shipit: 

Thanks!

**Pull request merged**